documents/coredocument.go

```diff
@@ -83,97 +86,79 @@ func newCoreDocument() (*CoreDocument, error) {
 		return nil, err
 	}
 
-	return &CoreDocument{cd}, nil
+	return &CoreDocument{Document: cd, CoreDocModified: true}, nil
 }
 
 // NewCoreDocumentFromProtobuf returns CoreDocument from the CoreDocument Protobuf.
 func NewCoreDocumentFromProtobuf(cd coredocumentpb.CoreDocument) *CoreDocument {
-	cd.EmbeddedDataSalts = nil
 	cd.EmbeddedData = nil
 	return &CoreDocument{Document: cd}
 }
 
-// NewCoreDocumentWithCollaborators generates new core Document with a document type specified by the prefix: po or invoice.
+// NewCoreDocumentWithCollaborators generates new core document with a document type specified by the prefix: po or invoice.
 // It then adds collaborators, adds read rules and fills salts.
 func NewCoreDocumentWithCollaborators(collaborators []string, documentPrefix []byte) (*CoreDocument, error) {
 	cd, err := newCoreDocument()
 	if err != nil {
 		return nil, errors.New("failed to create coredoc: %v", err)
 	}
-
 	ids, err := identity.NewDIDsFromStrings(collaborators)
 	if err != nil {
 		return nil, errors.New("failed to decode collaborators: %v", err)
 	}
 
 	cd.initReadRules(ids)
 	cd.initTransitionRules(ids, documentPrefix)
-	if err := cd.setSalts(); err != nil {
-		return nil, err
-	}
-
 	return cd, nil
 }
 
-// ID returns the Document identifier
+// SetDataModified sets the DataModified flag to the given value
+func (cd *CoreDocument) SetDataModified(modified bool) {
```

Comment:
remove

Reply:
What does this do?
Is this supposed to be removed ?